polarization/jonesmatrix.py

```diff
@@ -27,22 +27,60 @@ def __init__(self, A: complex = None,
         to avoid problems. For now, it is x and y.
         """
         if m is not None:
-            self.mOriginal = array(m)
+            self.mOriginal = m
         elif A is not None and D is not None and C is not None and D is not None:
-            self.mOriginal = array([[A,B],[C,D]])
+            self.mOriginal = [A,B,C,D]
         else:
             # Obviously, the subclass will compute when the time comes
             # See BirefringenceMaterial() for an example
             self.mOriginal = None
 
-        self.orientation = orientation
         self.L = physicalLength
 
+        self.orientation = orientation
+
         """ The basis vector for x and y. For now this is not really
         modifiable.  b1 x b2 = b3, direction of propagation """
-        self.b1 = Vector(1,0,0) # x̂
-        self.b2 = Vector(0,1,0) # ŷ
-        self.b3 = Vector(0,0,1) # ẑ
+        self.b1 = xHat  # Vector(1,0,0) # x̂
+        self.b2 = yHat  # Vector(0,1,0) # ŷ
+        self.b3 = zHat  # Vector(0,0,1) # ẑ
+
+    @property
+    def mOriented(self):
```

Comment:
The sign error has something to do with the orientation update. Maybe c*s should be -c*s. I don't know, this will require an orientation test on a more general matrix where B!=0 and C!=0.